iot-hub/Samples/service/FileUploadNotificationReceiverSample/FileUploadNotificationReceiverSample.cs

```diff
@@ -0,0 +1,128 @@
+
+        public FileUploadNotificationReceiverSample(string iotHubConnectionString, TransportType transportType, ILogger logger)
+        {
+            _iotHubConnectionString = iotHubConnectionString ?? throw new ArgumentNullException(nameof(iotHubConnectionString));
+            _transportType = transportType;
+            _logger = logger;
+        }
+
+        public async Task RunSampleAsync(TimeSpan runningTime)
+        {
+            using var cts = new CancellationTokenSource(runningTime);
+            Console.CancelKeyPress += (sender, eventArgs) =>
+            {
+                eventArgs.Cancel = true;
+                cts.Cancel();
+                _logger.LogInformation("Sample execution cancellation requested; will exit.");
+            };
+
+            try
+            {
+                await InitializeServiceClientAsync();
+                await ReceiveFileUploadNotifications(cts.Token);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError($"Unrecoverable exception caught, user action is required, exiting...: \n{ex}");
+            }
+        }
+
+        private async Task ReceiveFileUploadNotifications(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation($"Listening for file upload notifications from the service.");
+
+            FileNotificationReceiver<FileNotification> notificationReceiver = _serviceClient.GetFileNotificationReceiver();
+
+            int totalNotificationsReceived = 0;
+            int totalNotificationsCompleted = 0;
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    FileNotification fileUploadNotification = await notificationReceiver.ReceiveAsync(s_notificationReceiverTimeout);
+
+                    if (fileUploadNotification == null)
+                    {
+                        _logger.LogInformation($"Did not receive any notification after {s_notificationReceiverTimeout.TotalSeconds} seconds.");
+                        continue;
+                    }
+
+                    totalNotificationsReceived++;
+
+                    _logger.LogInformation($"Received file upload notification.");
+                    _logger.LogInformation($"\tDeviceId: {fileUploadNotification.DeviceId ?? "N/A"}.");
+                    _logger.LogInformation($"\tFileName: {fileUploadNotification.BlobName ?? "N/A"}.");
+                    _logger.LogInformation($"\tEnqueueTimeUTC: {fileUploadNotification.EnqueuedTimeUtc}.");
+                    _logger.LogInformation($"\tBlobSizeInBytes: {fileUploadNotification.BlobSizeInBytes}.");
+
+                    _logger.LogInformation($"Marking notification for {fileUploadNotification.DeviceId} as complete.");
+
+                    // Mark the notification as completed.
+                    await notificationReceiver.CompleteAsync(fileUploadNotification);
```

Comment:
It would also help if you added a small comment about what "completing" a notification means to a customer. How it makes it so the notification is removed from the service queue and won't be sent again